components/layout/layout_impl.rs

```diff
@@ -1155,6 +1166,26 @@ impl LayoutThread {
         self.stylist
             .force_stylesheet_origins_dirty(sheet_origins_affected_by_device_change);
     }
+
+    /// Main entry point for queries that require [PostCompositeQueryHelper].
+    fn process_post_composite_query<T>(
+        &self,
+        node: ServoLayoutNode<'_>,
+        process_query_internals: fn(
+            node: ServoLayoutNode<'_>,
+            helper: PostCompositeQueryHelper,
+        ) -> T,
+    ) -> T {
+        let stacking_context_tree = self.stacking_context_tree.borrow();
+
+        let helper = PostCompositeQueryHelper::new(
+            stacking_context_tree
+                .as_ref()
+                .map(|tree| &tree.compositor_info.scroll_tree),
+        );
+
+        process_query_internals(node, helper)
+    }
```

Comment:
Please remove this function as it is unnecessary. Instead let's create the `PostCompositeQueryHelper` in `query.rs`.

Reply:
Sure, it does seems overcomplicating the flow, thanks!